albatross/models/least_squares.h

```diff
@@ -103,39 +117,60 @@ class LeastSquaresRegression
  * Setup like this the resulting least squares solve will represent
  * an offset and slope.
  */
-using LinearRegressionBase =
-    AdaptedRegressionModel<double, LeastSquaresRegression>;
-
-class LinearRegression : public LinearRegressionBase {
+class LinearRegression : public LeastSquares<LinearRegression> {
 
 public:
-  LinearRegression(){};
-  std::string get_name() const override { return "linear_regression"; };
+  //  std::string get_name() const { return "linear_regression"; };
+
+  using Base = LeastSquares<LinearRegression>;
 
-  Eigen::VectorXd convert_feature(const double &x) const override {
+  Eigen::VectorXd convert_feature(const double &f) const {
     Eigen::VectorXd converted(2);
-    converted << 1., x;
+    converted << 1., f;
     return converted;
   }
 
+  std::vector<Eigen::VectorXd>
+  convert_features(const std::vector<double> &features) const {
+    std::vector<Eigen::VectorXd> output;
+    for (const auto &f : features) {
+      output.emplace_back(convert_feature(f));
+    }
+    return output;
+  }
+
+  Base::FitType fit(const std::vector<double> &features,
+           const MarginalDistribution &targets) const {
+    return Base::fit(convert_features(features),
+                     targets);
+  }
+
+  JointDistribution predict(const std::vector<double> &features,
+               const Base::FitType &least_squares_fit,
+               PredictTypeIdentity<JointDistribution> &&) const {
+    return Base::predict(convert_features(features),
+                         least_squares_fit,
+                         PredictTypeIdentity<JointDistribution>());
+  }
+
   /*
    * save/load methods are inherited from the SerializableRegressionModel,
    * but by defining them here and explicitly showing the inheritence
    * through the use of `base_class` we can make use of cereal's
    * polymorphic serialization.
    */
-  template <class Archive> void save(Archive &archive) const {
-    archive(cereal::make_nvp("linear_regression",
-                             cereal::base_class<LinearRegressionBase>(this)));
-  }
-
-  template <class Archive> void load(Archive &archive) {
-    archive(cereal::make_nvp("linear_regression",
-                             cereal::base_class<LinearRegressionBase>(this)));
-  }
+  //  template <class Archive> void save(Archive &archive) const {
```

Comment:
Does this commented code need to stick around?

Reply:
That'll be coming back soon ... but I'll remove it from this PR.